ui/web/src/modules/Copilot/Copilot.tsx

```diff
@@ -268,6 +298,9 @@ registerPage('Copilot', () => {
           const replaceMessages = (msgList: IChatMessage<any, any>[]) => {
             messages$.next(messages$.value.slice(0, idx + 1).concat(msgList));
           };
+          const editMessage = (payload: any) => {
+            msg.payload = payload;
```

Comment:
You should not modify the payload state in-place. React cannot detect the change.

Use messages$.next instead.